src/main/java/uk/gov/pay/directdebit/payments/api/PaymentRequestResponse.java

```diff
@@ -0,0 +1,124 @@
+
+    public PaymentRequestResponse(String paymentExternalId, Long amount, String returnUrl, String description, String reference, String createdDate, List<Map<String, Object>> dataLinks) {
+        this.paymentExternalId = paymentExternalId;
+        this.dataLinks = dataLinks;
+        this.amount = amount;
+        this.returnUrl = returnUrl;
+        this.description = description;
+        this.reference = reference;
+        this.createdDate = createdDate;
+    }
+
+    public List<Map<String, Object>> getDataLinks() {
+        return dataLinks;
+    }
+
+    public String getPaymentExternalId() {
+        return paymentExternalId;
+    }
+
+    public Long getAmount() {
+        return amount;
+    }
+
+    public String getReturnUrl() {
+        return returnUrl;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getReference() {
+        return reference;
+    }
+
+    public URI getLink(String rel) {
+        return dataLinks.stream()
+                .filter(map -> rel.equals(map.get("rel")))
+                .findFirst()
+                .map(link -> (URI) link.get("href"))
+                .get();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        PaymentRequestResponse that = (PaymentRequestResponse) o;
+
+        if (dataLinks != null ? !dataLinks.equals(that.dataLinks) : that.dataLinks != null) return false;
+        if (paymentExternalId != null ? !paymentExternalId.equals(that.paymentExternalId) : that.paymentExternalId != null) return false;
+        if (amount != null ? !amount.equals(that.amount) : that.amount != null) return false;
+        if (returnUrl != null ? !returnUrl.equals(that.returnUrl) : that.returnUrl != null) return false;
+        if (description != null ? !description.equals(that.description) : that.description != null) return false;
+        if (reference != null ? !reference.equals(that.reference) : that.reference != null) return false;
+        return createdDate != null ? createdDate.equals(that.createdDate) : that.createdDate == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = dataLinks != null ? dataLinks.hashCode() : 0;
+        result = 31 * result + (paymentExternalId != null ? paymentExternalId.hashCode() : 0);
+        result = 31 * result + (amount != null ? amount.hashCode() : 0);
+        result = 31 * result + (returnUrl != null ? returnUrl.hashCode() : 0);
+        result = 31 * result + (description != null ? description.hashCode() : 0);
+        result = 31 * result + (reference != null ? reference.hashCode() : 0);
+        result = 31 * result + (createdDate != null ? createdDate.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        // Some services put PII in the description, so don’t include it in the stringification
```

Comment:
Rather than a comment I would add a test that specifically check that description field is not present when doing toString (with proper named method).